### ISSUE TITLE
Integrating with a bound codebase which already uses ImGUI

### ISSUE THREAD

**Issue by nmwsharp** [swistakm/pyimgui#215]
**TL;DR**: Is it possible to use these bindings when the ImGUI context is already being initialized and used elsewhere?

I manage a [Python codebase](https://polyscope.run/py/) which is essentially just pybind11 bindings to a large C++ project, and that C++ project already uses ImGUI extensively internally. I'm looking to enable users of my Python bindings to issue ImGUI commands to extend the UI (adding their own buttons in a callback, etc). I'd love to leverage pyimgui's bindings rather than writing my own from scratch.  

This is a bit unusual, since I don't want pyimgui to initialize and manage a context, the context will already be initialized elsewhere in bound code. For the sake simplicity, let's assume pyimgui and my library are built against the exact same version of ImGUI. 

Is this something that could potentially be done? For instance, could I pass in a pointer to an ImGUI context in lieu of normal initialization? Do you know of any examples where this has been done?

Thanks for the great library!

**Comment by learn-more** [comment]
There is `get_current_context`, so the fastest way to find this out is probably to try it?

**Comment by nmwsharp** [comment]
Ah, thanks for the pointer! I see that there's a matching `set_current_context()`. I can play around with it and see what seems to work.

(although, even if that works it would be nice to know if this is a supported use-case for pyimgui and is likely to continue to work in the future 😄)
